src/command_graph_generator.cc

```diff
@@ -53,35 +53,18 @@ void command_graph_generator::notify_host_object_destroyed(const host_object_id
 	m_host_objects.erase(hoid);
 }
 
-// According to Wikipedia https://en.wikipedia.org/wiki/Topological_sorting#Depth-first_search
-// TODO: This may no longer be necessary since different command types are now generated in pre-determined order - revisit
-std::vector<abstract_command*> sort_topologically(command_set unmarked) {
-	command_set temporary_marked;
-	command_set permanent_marked;
-	std::vector<abstract_command*> sorted(unmarked.size());
-	auto sorted_front = sorted.rbegin();
-
-	const auto visit = [&](abstract_command* const cmd, auto& visit /* to allow recursion in lambda */) {
-		if(permanent_marked.count(cmd) != 0) return;
-		assert(temporary_marked.count(cmd) == 0 && "cyclic command graph");
-		unmarked.erase(cmd);
-		temporary_marked.insert(cmd);
-		for(const auto dep : cmd->get_dependents()) {
-			visit(dep.node, visit);
+/// Returns whether an iterator range of commands is topologically sorted, i.e. sequential execution would satisfy all internal dependencies.
+template <typename Iterator>
+bool is_topologically_sorted(Iterator begin, Iterator end) {
```

Comment:
Minor: Could be scoped inside `command_graph_generator` or be made static (The IDAG function of the same name is inside `instruction_graph_generator_detail` to show it isn't truly generic).

Reply:
In the interest of the planet I'll avoid another CI run. Medium-term I'd like to align the CGGEN with the newer interfaces to use a PIMPL and move the implementation to `command_graph_generator_detail`.